libid/fractalb.cpp

```diff
@@ -1072,6 +1073,40 @@ julia_z_power_bf_fractal()
     return g_bailout_bigfloat();
 }
 
+int
+burning_ship_bf_fractal()
+{
+    int degree = (int) g_params[2];
+    BFComplex big_z;
+    int saved;
```

Comment:
See `biginit.h` and use the `BigStackSaver` to save/restore the big number stack safely.  Localize the use of `BigStackSaver` to the `else` block where you allocate space.

Reply:
Good call. Done.